### PR TITLE
Improve testing of the RefreshIndicator widget.

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: packages/flutter/test/material/refresh_indicator_test.dart [hunk 1]
```diff
@@ -41,4 +47,179 @@ void main() {
...
+
+  testWidgets('RefreshIndicator - not enough', (WidgetTester tester) async {
+    refreshCalled = false;
+    await tester.pumpWidget(
+      new RefreshIndicator(
+        scrollableKey: scrollableKey,
+        refresh: refresh,
+        child: new Block(
+          scrollableKey: scrollableKey,
+          children: <String>['A'].map((String item) {
```
Comment (HansMuller) [inline packages/flutter/test/material/refresh_indicator_test.dart]: Why not just write
```
children: <Widget>[
  new SizedBox(height: 200.0, child: new Text('A')),
]
```

**Comment by abarth** [comment]
LGTM

File: packages/flutter/test/material/refresh_indicator_test.dart [hunk 2]
```diff
@@ -41,4 +47,179 @@ void main() {
...
+            return new SizedBox(
+              height: 200.0,
+              child: new Text(item)
+            );
+          }).toList()
+        )
+      )
+    );
+
+    await tester.fling(find.text('A'), const Offset(0.0, 100.0), -1000.0);
```
Comment (HansMuller) [inline packages/flutter/test/material/refresh_indicator_test.dart]: I'm not sure what this test is supposed to check. The title implies that that the fling hasn't been dragged far enough, but it looks like the fling is going in the wrong direction?

File: packages/flutter/test/material/refresh_indicator_test.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/test/material/refresh_indicator_test.dart]: Well, now that you mention it...

**Comment by HansMuller** [comment]
LGTM too.

**Comment by Hixie** [comment]
Landing when green.

This now includes some changes to tab code.

**Pull request merged**